File: hacker_rank/algorithms/strings/two_characters.py
#!/bin/python3

import sys

def is_t(arr):
    for i in range(0, len(arr) - 2):
        if arr[i] != arr[i + 2] or arr[i] == arr[i+1]:
            return False
    return True

s_len = int(input().strip())
s = list(input().strip())
letters = {}
for a in s:
    letters[a] = True
letters = list(letters.keys())
largest = 0
for a in letters:
    for b in letters:
        if a == b:
            continue
            
        arr = [c for c in s if c == a or c == b]
        largest = len(arr) if len(arr) > largest and is_t(arr) else largest

print(largest)

File: hacker_rank/algorithms/strings/two_characters.rb
#!/bin/ruby

def is_t(arr)
    (0...arr.size - 2).each do |i|
        return false unless arr[i] != arr[i+1]
        return false unless arr[i] == arr[i+2]
    end
end

len = gets.strip.to_i
s = gets.strip.chars
largest = 0
letters = Hash.new(0)
s.each { |a| letters[a] += 1 }
max = 0
letters = letters.keys
letters.each do |a|
    letters.each do |b|
        next if a == b
        arr = s.select { |c| c == a || c == b }
        max = arr.size if is_t(arr) && arr.size > max
    end
end
puts max
